pkg/api/kubegraph/analysis/hpa.go

```diff
@@ -0,0 +1,66 @@
+package analysis
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/apis/extensions"
+
+	osgraph "github.com/openshift/origin/pkg/api/graph"
+	kubegraph "github.com/openshift/origin/pkg/api/kubegraph/nodes"
+)
+
+const (
+	HPAMissingCPUTargetError      = "HPAMissingCPUTarget"
+	HPAOverlappingScaleRefWarning = "HPAOverlappingScaleRef"
+)
+
+func FindHPASpecsMissingCPUTargets(g osgraph.Graph, f osgraph.Namer) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+	for _, uncastNode := range g.NodesByKind(kubegraph.HorizontalPodAutoscalerKind) {
+		node := uncastNode.(*kubegraph.HorizontalPodAutoscalerNode)
+
+		hpa, ok := node.Object().(*extensions.HorizontalPodAutoscaler)
+		if !ok {
+			continue
+		}
+
+		if hpa.Spec.CPUUtilization == nil {
+			markers = append(markers, osgraph.Marker{
+				Node:     node,
+				Severity: osgraph.ErrorSeverity,
+				Key:      HPAMissingCPUTargetError,
+				Message:  fmt.Sprintf("horizontal pod autoscaler %q is missing a CPU utilization target", hpa.Name),
```

Comment:
no need to mention the whole resource name. this should be handled just by printing the name of the node (and should be hpa)